Bluejay/Bluejay/CharacteristicIdentifier.swift

```diff
@@ -40,19 +40,24 @@ public struct CharacteristicIdentifier: Hashable {
         self.service = service
     }
     
-    /// The hash value of the `CBUUID`.
-    public var hashValue: Int {
-        return uuid.hashValue
+    /// Check equality between a `CharacteristicIdentifier` and a `CBCharacterstic`.
+    public static func ==(lhs: CharacteristicIdentifier, rhs: CBCharacteristic) -> Bool {
+        return (lhs.uuid == rhs.uuid) && (lhs.service.uuid == rhs.service.uuid)
     }
-    
+}
+
+extension CharacteristicIdentifier: Equatable {
+
     /// Check equality between two CharacteristicIdentifiers.
     public static func ==(lhs: CharacteristicIdentifier, rhs: CharacteristicIdentifier) -> Bool {
         return (lhs.uuid == rhs.uuid) && (lhs.service.uuid == rhs.service.uuid)
     }
-    
-    /// Check equality between a `CharacteristicIdentifier` and a `CBCharacterstic`.
-    public static func ==(lhs: CharacteristicIdentifier, rhs: CBCharacteristic) -> Bool {
-        return (lhs.uuid == rhs.uuid) && (lhs.service.uuid == rhs.service.uuid)
+}
+
+extension CharacteristicIdentifier: Hashable {
+
+    /// The hash value of the `CBUUID`.
+    public var hashValue: Int {
+        return uuid.hashValue
```

Comment:
We probably should do something slightly different here, as this will mean two characteristics with the same UUID but on different services will hash the same. 

It might not be a problem in practice, I don't THINK that you are allowed to have two different services that have characteristics with the same UUID, but I'm not 100% confident in that, and even if that isn't allowed, having them has the same could in theory cause issues with malformed characteristic identifiers (if you create a characteristic with the WRONG service identifier and then try to use it in a Set or something alongside ones that are one the correct service you could get weird results)

So we need to combine the hash values for the characteristic and service such as in: https://stackoverflow.com/questions/42627014/implementing-a-hash-combiner-in-swift
 So a reasonable implementation here would be:

```
let c = uuid.hashValue
let s = service.hashValue
return (c << 5) &+ c &+ s
```

Reply:
done